src/InstrumentationEngine/Instruction.h

```diff
@@ -137,18 +138,40 @@ namespace MicrosoftInstrumentationEngine
         HRESULT SetInstructionGeneration(_In_ InstructionGeneration instructionGeneration);
 
         static HRESULT InstructionFromBytes(_In_ LPCBYTE pCode, _In_ LPCBYTE pEndOfCode, _Out_ CInstruction** ppInstruction);
-        static HRESULT OrdinalOpcodeFromBytes(_In_ LPCBYTE pCode, _In_ LPCBYTE pEndOfCode, _Out_ ILOrdinalOpcode* pOpcode);
+        static HRESULT OrdinalOpcodeFromBytes(_In_reads_to_ptr_(pEndOfCode) LPCBYTE pCode, _In_ LPCBYTE pEndOfCode, _Out_ ILOrdinalOpcode* pOpcode);
 
 
-        HRESULT EmitIL(_In_ BYTE* pILBuffer, _In_ DWORD dwcbILBuffer);
+        HRESULT EmitIL(_In_reads_bytes_(dwcbILBuffer) BYTE* pILBuffer, _In_ DWORD dwcbILBuffer);
 
-        HRESULT LogInstruction(_In_ BOOL ignoreTest);
+        HRESULT LogInstruction(bool ignoreTest);
 
         constexpr CInstruction* NextInstructionInternal() { return m_pNextInstruction.p; }
         constexpr CInstruction* PreviousInstructionInternal() { return m_pPreviousInstruction.p; }
         constexpr CInstruction* OriginalNextInstructionInternal() { return m_pOriginalNextInstruction.p; }
         constexpr CInstruction* OriginalPreviousInstructionInternal() { return m_pOriginalPreviousInstruction.p; }
 
+    protected:
+        template<typename... Args>
+        HRESULT QueryInterfaceInternal(_In_ REFIID riid, _Out_ void** ppvObject, Args... pThisArgs)
```

Comment:
>QueryInterfaceInternal [](start = 16, length = 22)

This function seems not necessary. You could just put the if check inside QueryInterface, then you wouldn't have to forward the args template.